src/core/login/pin2.ts

```diff
@@ -173,14 +173,18 @@ export async function changePin(
 export async function checkPin2(
   ai: ApiInput,
   login: LoginTree,
-  pin: string
+  pin: string,
+  forDuressAccount?: boolean
 ): Promise<boolean> {
-  const { appId, loginId, username } = login
+  const { loginId, username } = login
   if (username == null) return false
 
+  const appId =
+    forDuressAccount === true ? login.appId + '.duress' : login.appId
```

Comment:
If we are already in duress mode, the appId would be `.duress.duress`.

Reply:
Ah, actually `login` is always `loginTree` (a root). If it's not a root, then `if (username == null) return false` would prevent this line from being reached. Unless I misunderstand that non-roots can have a username field? If so, then it's potentially non-root, and `.duress.duress` is possible, but then the stash search would fail, and we'd get an exception out of this function. So generally this is safe.

You did make me think of another test case. I added it and included changes.